### PR TITLE
chef client is successfully reading from the windows cert store and a…

### PR DESCRIPTION
**Pull request by johnmccrae**
…ttempting to connect to the chef server

Signed-off-by: John McCrae <john.mccrae@progress.com>

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Chore (non-breaking change that does not add functionality or fix an issue)

## Checklist:

- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have run the pre-merge tests locally and they pass.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [ ] All commits have been signed-off for [the Developer Certificate of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

### CODE REVIEW

File: lib/chef/client.rb [hunk 1]
```diff
@@ -638,13 +642,21 @@ def node_name
     # @api private
     #
     def register(client_name = node_name, config = Chef::Config)
+      puts "\n cli.rb - This is the client name I am using : #{client_name}"
+      puts "\n cli.rb - This is my client key : #{config[:client_key]}"
       if !config[:client_key]
         events.skipping_registration(client_name, config)
         logger.trace("Client key is unspecified - skipping registration")
-      elsif File.exists?(config[:client_key])
+      elsif detect_certificate_key(client_name)
```
Comment (lamont-granquist) [inline lib/chef/client.rb:650]: change this to `Chef::Http::Authenticator.detect_certificate_key(client_name)` and move it all to the other class and declare that as `def self.detect_certificate_key(client_name)`

Comment (lamont-granquist) [reply lib/chef/client.rb:650]: down in the authenticator class you'll probably want to create a method to delegate an instance method to the class method:

```
def detect_certificate_key(client_name)
  self.class.detect_certificate_key(client_name)
end

def self.detect_certificate_key(client_name)
[...]
end
```

that lets you call `detect_certificate_key(client_name)` from other methods in the authenticator class without having to remember to call the class method

**Comment by marcparadise** [comment]
This needs a cleanup/rebase; can't easily determine which of these changes are yours; It looks like a rebase was done, but possibly with the wrong base.

**Comment by johnmccrae** [comment]
Code merged into 17.10.0

**Pull request closed without merging**